### PR TITLE
Convert plugin to new node-based API

### PR DESCRIPTION
**Pull request by lmarburger**
### Todo

- [x] Provide a search string
- [x] Tail

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+var Heroku = require('heroku-client');
+var https = require('https');
+var strftime = require('strftime');
+
+exports.topics = [{
+  name: "pt2",
+  description: "Search and/or tail logs (optional: tail; search query)"
```
Comment (troy) [inline index.js]: Worth tweaking the description? I could see "Search and/or tail logs"

File: index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  description: "Search and/or tail logs (optional: tail; search query)",
+  help: "Search and/or tail logs (optional: tail; search query)",
+  needsApp: true,
+  needsAuth: true,
+  run: function (context) {
+    var heroku = new Heroku({token: context.auth.password});
+    heroku.apps(context.app).configVars().info(function (err, config) {
+      if (err) { throw err; }
+      if (!config.PAPERTRAIL_API_TOKEN) {
+        console.error('Add the Papertrail addon to this application: https://addons.heroku.com/papertrail');
```
Comment (troy) [inline index.js]: Nice check. Grammar nit: "add-on"

File: index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      var options = {
+        host: 'papertrailapp.com',
+        path: '/api/v1/events/search.json',
+        port: '443',
+        headers: {'X-Papertrail-Token': config.PAPERTRAIL_API_TOKEN}
+      };
+
+      var printEvent = function(event) {
+        var received_at = strftime("%b %e %T", new Date(event.received_at));
```
Comment (troy) [inline index.js]: I could see taking this chance to move to ISO 8601 time for display, like:

```2015-02-04T17:40:41+00:00```

`received_at` contains all of this, I believe.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    "type": "git",
+    "url": "https://github.com/papertrail/papertrail-heroku-plugin"
+  },
+  "bugs": {
+    "url": "https://github.com/papertrail/papertrail-heroku-plugin/issues"
+  },
+  "keywords": [
+    "heroku-plugin",
+    "papertrail",
+    "log"
```
Comment (troy) [inline package.json]: I'd propose `logs`, `logging`, and `papertrailapp` as well. I just [checked](https://www.npmjs.com/search?q=logs) and substrings aren't automatically matched.

**Comment by lmarburger** [comment]
Good feedback. Just pushed all the grammar edits in 69c759f Will play with the date format in a later commit.

File: index.js (hunk 3 above)
Comment (lmarburger) [inline index.js]: ed2c727 fixes this

**Comment by lmarburger** [comment]
This is ready for release whenever Heroku and us are ready!

**Comment by lmarburger** [comment]
@dickeyxxx What are your thoughts for releasing this? As I understand it, installing a plugin from npm is supported in the latest releases. @troy tried the other day and it failed presumably because he has an ancient version of the cli/toolbelt. Should we just add a note to the top of the readme to link to upgrading the toolbelt?

That's the last question I have before this is ready to be :ship:

**Comment by dickeyxxx** [comment]
I assume he did have an old version, this stuff is really new of course. I think it's ready for mainstream use right now. Being so new I do anticipate a bug or two we will have to fix, but nothing major.

**Comment by lmarburger** [comment]
OK great. I'm going to see if I can get my hands on an old version of the toolbox/CLI to see what error message someone will see if they try to install a node plugin where it's not supported.

**Comment by lmarburger** [comment]
@dickeyxxx Just [published version 1.0.0](https://www.npmjs.com/package/heroku-papertrail), merged this branch into master, and [moved the old ruby code](https://github.com/papertrail/papertrail-heroku-plugin/tree/deprecated-ruby) to its own branch. :tada:

Thanks for the help!

**Pull request merged**